app/services/TwoStepVerification.scala

```diff
@@ -44,10 +47,34 @@ trait TwoStepVerification {
 
   def biz2svRules: List[Biz2SVRule]
 
+  def auditConnector: AuditConnector
+
   lazy val upliftLocations = upliftLocationsConfiguration.map(s => s.split(",").map(_.trim).map(stringToLocation).toSet).getOrElse(Set.empty)
 
   def isUplifted(location: Location) = upliftLocations.contains(location)
 
+  private def sendAuditEvent(biz2SVRule: Biz2SVRule, ruleContext: RuleContext, mandatory: Boolean)(implicit authContext: AuthContext, request: Request[AnyContent], hc: HeaderCarrier) = {
+    val enrolmentsFut = ruleContext.activeEnrolments
+    val userDetailsFut = ruleContext.userDetails
+    for {
+      enrolments <- enrolmentsFut
+      userDetails <- userDetailsFut
+    } yield {
+      val auditEvent = ExtendedDataEvent(
+        auditSource = AppName.appName,
+        auditType = "TwoStepVerificationOutcome",
+        tags = hc.toAuditTags("no two step verification", request.path),
```

Comment:
the value of transaction name should be depending on the fact 2SV is mandatory or optional:
- optional: "two step verification optional"
- mandatory: "two step verification mandatory"